client/ayon_tvpaint/lib.py

```diff
@@ -515,8 +517,24 @@ def fill_reference_frames(frame_references, filepaths_by_frame):
         src_filepath = filepaths_by_frame[ref_idx]
         dst_filepath = filepaths_by_frame[frame_idx]
 
-        if hasattr(os, "link"):
-            os.link(src_filepath, dst_filepath)
+        # This is to avoid errors on Windows when too many hardlinks are
+        # created with longer sequences (more than 1024). We fall back to
+        # copy in that case.
+
+        if hasattr(os, "link") and _can_link:
+            if platform.system().lower() == "windows":
+                try:
+                    os.link(src_filepath, dst_filepath)
+                except OSError as e:
+                    if getattr(e, 'winerror', None) != 1142:
+                        raise
+                    # An attempt was made to create more links on a file
+                    # than the file system supports
+                    #   - fallback to copy
+                    shutil.copy(src_filepath, dst_filepath)
+                    _can_link = False
+            else:
+                os.link(src_filepath, dst_filepath)
         else:
             shutil.copy(src_filepath, dst_filepath)
```

Comment:
```suggestion
        if not hasattr(os, "link"):
            shutil.copy(src_filepath, dst_filepath)
            continue
        if platform.system().lower() != "windows":
            os.link(src_filepath, dst_filepath)
            continue
        try:
            os.link(src_filepath, dst_filepath)
        except OSError as e:
            if getattr(e, 'winerror', None) != 1142:
                raise
            # An attempt was made to create more links on a file
            # than the file system supports
            #   - fallback to copy
            shutil.copy(src_filepath, dst_filepath)
```
It is not working with 1 file. Just because 1 file cannot be linked anymore doesn't mean the others can't be.